backend/src/routes/api/users.routes.js

```diff
@@ -0,0 +1,37 @@
+import { StatusCodes } from 'http-status-codes';
+import User, { userValidator } from '../../models/User.js';
+import ExpressError from '../../middlewares/ExpressError.js';
+
+const userEndpoint = (router) => {
+  router.patch('/users/:id', getUser, userValidator, patchUser);
+
+  async function getUser(req, res, next) {
+    let user;
+    try {
+      user = await User.findById(req.params.id);
+      if (user == null) {
+        throw new ExpressError("Can't find a user", StatusCodes.NOT_FOUND);
+      }
+    } catch (err) {
+      next(err);
+    }
+
+    res.user = user;
+    next();
+  }
```

Comment:
let's extract it from this function - please move it to eg. scr/middlewares/userMiddlewares.js

Reply:
done